quipucords/tests/scanner/openshift/test_api.py

```diff
@@ -178,6 +186,137 @@ def test_retrieve_operators(ocp_client: OpenShiftApi):
     assert isinstance(operators[-1], LifecycleOperator)
 
 
+@pytest.mark.vcr(
+    VCRCassettes.OCP_ACM,
+    VCRCassettes.OCP_DISCOVERER_CACHE,
+)
+def test_retrieve_acm_metrics(ocp_client: OpenShiftApi):
+    """Test retrieve acm metrics method."""
+    # OCP instance used on VCR preparation should have ACM operator installed,
+    # with active multiClusterHub instance
+    acm_metrics = ocp_client.retrieve_acm_metrics()
+    assert isinstance(acm_metrics, list)
+
+    # The following assertion is based on the assumption that the provisioned cluster
+    # has not undergone any additional config modifications. When creating the
+    # multiClusterHub instance, it is expected that at least one managed cluster will
+    # be discovered, the hub itself. An associated set of keys is expected.
+    # The "available" key within these metrics is expected to be True, indicating that
+    # the cluster is available and configured properly.
+
+    expected_keys = [
+        "vendor",
+        "cloud",
+        "version",
+        "managed_cluster_id",
+        "available",
+        "core_worker",
+        "socket_worker",
+        "created_via",
+    ]
+    expected_available_value = True
+
+    found_matching_dict = False
+    for managed_cluster in acm_metrics:
+        if all(key in managed_cluster for key in expected_keys):
+            if managed_cluster["available"] == expected_available_value:
+                found_matching_dict = True
+                break
+    assert found_matching_dict
```

Comment:
Question about the data: why only 1 out of 4 clusters is available on the setup?

Regardless, here's a cleaner way to check the data and test ALL the output (not only the "available" cluster should have the expected facts)
```suggestion
    expected_data = [{key: mock.ANY for key in expected_keys}] * len(acm_metrics)
    assert acm_metrics == expected_data
    # at least one cluster is available
    assert any(cluster_metrics["available"] for cluster_metrics in acm_metrics)
```

Reply:
When I wrote this I was thinking about not expecting extra clusters in case we would have to rewrite the vcr cassettes. But what you proposed above works really well to check the expected facts regardless of the numbers of clusters present. we should guarantee that all the expected_facts are there for every single cluster + at least one cluster which available key is True.